### PR TITLE
Feat/lesq 515/specialist subject listing query

### PR DESCRIPTION
**Pull request by njblore**
## Description

Music year: 1984

- adds queries for the specialist subject listing page
- adds page for specialist subject listing
- update types 
- add 404 fallback when no specialist content is published

## How to test

1. Go to https://deploy-preview-2294--oak-web-application.netlify.thenational.academy/teachers/specialist/subjects
3. when there is no specialist data published there will be a 404 page (broken on preview so will `internal server error`)

## Screenshots

How it should now look:

### CODE REVIEW

File: src/pages/teachers/specialist/subjects.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    getProps: async () => {
+      const curriculumData = await curriculumApi2023.specialistSubjectListing();
+
+      if (!curriculumData) {
+        return {
+          notFound: true,
+        };
+      }
+
+      // TODO: remove this once specialist is published
```
Comment (JBR90) [inline src/pages/teachers/specialist/subjects.tsx:50]: Is there a ticket for this?

**Review by JBR90 — APPROVED** [review]
Looks good to me, one comment.

**Review by JBR90 — APPROVED** [review]
Looks good to me, one comment.

File: src/pages/teachers/specialist/subjects.tsx (hunk 1 above)
Comment (njblore) [reply src/pages/teachers/specialist/subjects.tsx:50]: no, good shout! I'll add one 👍

**Review by jack-skerrett-oak — APPROVED** [review]
Going to say the classic "all seems fine"... let's hope it is :D

**Pull request merged**